inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -215,68 +212,102 @@ def arxiv_plot_extract(obj, eng):
+    builder = LiteratureBuilder()
+    if isinstance(xml_content, str):
+        xml_content = xml_content.decode('utf-8')
 
-            obj.log.info('Extracted tarball to: {0}'.format(scratch_space))
-            xml_files_list = [path for path in file_list if path.endswith('.xml')]
-            obj.log.info('Found xmlfiles: {0}'.format(xml_files_list))
-
-            extracted_authors = []
-            for xml_file in xml_files_list:
-                with open(xml_file, 'r') as xml_file_fd:
-                    xml_content = xml_file_fd.read()
-
-                match = REGEXP_AUTHLIST.findall(xml_content)
-                if match:
-                    obj.log.info('Found a match for author extraction')
-                    try:
-                        authors_xml = convert(xml_content, stylesheet)
-                    except XMLSyntaxError:
-                        # Probably the %auto-ignore comment exists, so we skip the
-                        # first line. See: inspirehep/inspire-next/issues/2195
-                        authors_xml = convert(
-                            xml_content.split('\n', 1)[1],
-                            stylesheet,
-                        )
-
-                    extracted_authors.extend(marcxml2record(authors_xml).get('authors', []))
-
-            if extracted_authors:
-                for author in extracted_authors:
-                    author['full_name'] = decode_latex(author['full_name'])
-
-                obj.data['authors'] = extracted_authors
-                set_mark(obj, "authors_xml", True)
+    # Probably the %auto-ignore comment exists, so we skip the
+    # first line. See: inspirehep/inspire-next/issues/2195
+    if "%auto-ignore" in xml_content:
+        xml_content = xml_content.split('\n', 1)[1]
+
+    content = Selector(text=xml_content, type="xml")
+
+    content.remove_namespaces()
+    # Going through all the authors in the file
+    for author in content.xpath("//Person"):
+
+        ids = []
+        affiliations = []
+
+        # Getting all the author ids
+        for source, id in itertools.izip(author.xpath('./authorIDs/authorID[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/@source | ./authorids/authorid[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/@source').getall(), author.xpath('./authorIDs/authorID[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/text() | ./authorids/authorid[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/text()').getall()):
```

Comment:
maybe it would be easier to do all the string filtering in python rather than xpath?

Reply:
I'm doing now most of the filtering in python. I'm still checking the empty ones in xpath, otherwise, the order in the list might not be correct and the number of items in source and id might differ.

Reply:
that's a good point, I hadn't thought of that. But wouldn't it be the other way around? what if you have something like
`<authorID source="INSPIRE"></authorID>`. Wouldn't that mess up the alignement?

Reply:
Yes. That's why I'm checking the source and the value in both of the XPath() searches  ```/authorID[@source!="" and text()!=""]/@source```
```/authorID[@source!="" and text()!=""]/text()```

Reply:
:facepalm: of course, you're right, I didn't look carefully enough